src/main/webapp/app/shared/metis/posting-reactions-bar/posting-reactions-bar.component.ts

```diff
@@ -148,30 +150,32 @@ export abstract class PostingsReactionsBarDirective<T extends Posting> implement
     }
 
     /**
-     * builds the ReactionCountMap data structure out of a given array of reactions
-     * @param reactions array of reactions associated to the current posting
-     */
-    buildEmojiIdCountMap(reactions: Reaction[]): ReactionCountMap {
-        return reactions.reduce((countMap: ReactionCountMap, reaction: Reaction) => {
-            const hasReacted = reaction.user?.id === this.metisService.getUser().id;
-            const reactionCount = {
-                count: countMap[reaction.emojiId!] ? countMap[reaction.emojiId!].count + 1 : 1,
-                hasReacted: countMap[reaction.emojiId!] ? countMap[reaction.emojiId!].hasReacted || hasReacted : hasReacted,
-            };
-            return { ...countMap, [reaction.emojiId!]: reactionCount };
-        }, {});
-    }
-
-    /**
-     * updates the posting's reactions by calling the build function for the reactionCountMap if there are any reaction on the posting
+     * updates the posting's reactions by calling the build function for the reactionMetaDataMap if there are any reaction on the posting
      */
     updatePostingWithReactions(): void {
         if (this.posting.reactions && this.posting.reactions.length > 0) {
-            // filter out emoji for pin and archive as they should not be listed in the reactionCountMap
+            // filter out emoji for pin and archive as they should not be listed in the reactionMetaDataMap
             const filteredReactions = this.posting.reactions.filter((reaction: Reaction) => reaction.emojiId !== this.pinEmojiId || reaction.emojiId !== this.archiveEmojiId);
-            this.reactionCountMap = this.buildEmojiIdCountMap(filteredReactions);
+            this.reactionMetaDataMap = this.buildReactionMetaDataMap(filteredReactions);
         } else {
-            this.reactionCountMap = {};
+            this.reactionMetaDataMap = {};
         }
     }
+
+    /**
+     * builds the ReactionMetaDataMap data structure out of a given array of reactions
+     * @param reactions array of reactions associated to the current posting
+     */
+    buildReactionMetaDataMap(reactions: Reaction[]): ReactionMetaDataMap {
+        return reactions.reduce((metaDataMap: ReactionMetaDataMap, reaction: Reaction) => {
+            const hasReacted = reaction.user?.id === this.metisService.getUser().id;
+            const reactingUser = hasReacted ? PLACEHOLDER_USER_REACTED : reaction.user?.name!;
+            const reactionMetaData: ReactionMetaData = {
+                count: metaDataMap[reaction.emojiId!] ? metaDataMap[reaction.emojiId!].count + 1 : 1,
+                hasReacted: metaDataMap[reaction.emojiId!] ? metaDataMap[reaction.emojiId!].hasReacted || hasReacted : hasReacted,
+                reactingUsers: metaDataMap[reaction.emojiId!] ? metaDataMap[reaction.emojiId!].reactingUsers.concat(reactingUser) : [reactingUser],
```

Comment:
[nit] Suggestions for these lines (just a bit simpler imo, you can leave it if you deem the current version to be better or equal). :)
```suggestion
                count: 1 + (metaDataMap[reaction.emojiId!]?.count || 0),
                hasReacted: metaDataMap[reaction.emojiId!]?.hasReacted || hasReacted,
                reactingUsers: [...(metaDataMap[reaction.emojiId!]?.reactingUsers || []), reactingUser],
```

Reply:
thanks for the suggestion 👍  I would leave it as is, because we tested this version now :) If I need to change something else, I can reconsider your suggestions. Otherwise, it will be merged with the current version.